src/Controller/Admin/ModuleSelectionController.php

```diff
@@ -0,0 +1,112 @@
+use PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController;
+use PrestaShopBundle\Security\Annotation\AdminSecurity;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\ServiceUnavailableHttpException;
+use Symfony\Component\Templating\EngineInterface;
+
+/**
+ * Responsible of "Improve > Modules > Modules Catalog" page display.
+ */
+class ModuleSelectionController extends FrameworkBundleAdminController
+{
+    /**
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller'))")
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function indexAction(Request $request)
+    {
+        $response = new Response();
+
+        try {
+            $externalContentProvider = new ExternalContentProvider();
+
+            $content = $this->getTemplateEngine()->render(
+                '@Modules/ps_mbo/views/templates/admin/controllers/module_catalog/addons_store.html.twig',
+                [
+                    'pageContent' => $externalContentProvider->getContent($this->getAddonsUrl($request)),
+                    'layoutHeaderToolbarBtn' => [],
+                    'layoutTitle' => $this->trans('Module selection', 'Admin.Navigation.Menu'),
+                    'requireAddonsSearch' => true,
+                    'requireBulkActions' => false,
+                    'showContentHeader' => true,
+                    'enableSidebar' => true,
+                    'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
+                    'requireFilterStatus' => false,
+                    'level' => $this->authorizationLevel($request->attributes->get('_legacy_controller')),
+                ]
+            );
+            $response->setContent($content);
+        } catch (ServiceUnavailableHttpException $exception) {
+            $response->setContent($this->getTemplateEngine()->render('@Modules/ps_mbo/views/templates/admin/error.html.twig'));
+            $response->setStatusCode($exception->getStatusCode());
+            $response->headers->add($exception->getHeaders());
+        }
+
+        return $response;
+    }
+
+    /**
+     * @param Request $request
+     *
+     * @return string
+     */
+    private function getAddonsUrl(Request $request)
+    {
+        $psVersion = $this->get('prestashop.core.foundation.version')->getVersion();
+        $parent_domain = $request->getSchemeAndHttpHost();
+        $context = $this->getContext();
+        $currencyCode = $context->currency->iso_code;
+        $languageCode = $context->language->iso_code;
+        $countryCode = $context->country->iso_code;
+        $activity = $this->get('prestashop.adapter.legacy.configuration')->getInt('PS_SHOP_ACTIVITY');
+
+        return "https://addons.prestashop.com/iframe/search-1.7.php?psVersion=$psVersion"
```

Comment:
```suggestion
        return 'https://addons.prestashop.com/iframe/search-1.7.php?psVersion=' . $psVersion
```

Reply:
And idem for the string

Reply:
Now handled by AddonsSelectionLinkProvider service